Sources/AccountSDKIOSWeb/User/User.swift

```diff
@@ -145,24 +147,7 @@ extension User {
             case .failure(.errorResponse(let code, let body)):
                 // 401 might indicate expired access token
                 if code == 401 {
-                    self.client.refreshTokens(for: self) { result in
-                        switch result {
-                        case .success(_):
-                            // retry the request with fresh tokens
-                            self.makeRequest(request: request, completion: completion)
-                        case .failure(.refreshRequestFailed(.errorResponse(_, let body))):
-                            guard User.shouldLogout(tokenResponseBody: body) else {
-                                completion(requestResult)
-                                return
-                            }
-
-                            SchibstedAccountLogger.instance.info("Invalid refresh token, logging user out")
-                            self.logout()
-                            completion(.failure(.unexpectedError(underlying: LoginStateError.notLoggedIn)))
-                        case .failure(_):
-                            completion(requestResult)
-                        }
-                    }
+                    self.refreshHandler.startRefresh(user: self, requestResult: requestResult, request: request, completion: completion)
```

Comment:
As of this commit https://github.com/schibsted/AccountSDKIOSWeb/commit/6c04c690d85524c365f086384e1652c24682a06a#diff-6987f304cb712e82d9dbe50819454420e008f39103e82b95bd25979c220c82ba
this function on the User is no longer public, so I assume the refreshHandler should also (or only, if this function is being removed) be applied to the `AuthenticatedUrlSession` (it might make sense to align the file and class name to avoid confusion, as the file is called `URLSessionWithUser` https://github.com/schibsted/AccountSDKIOSWeb/blob/640de76c9013232a04844c4e0e1712e7e11cc8a1/Sources/AccountSDKIOSWeb/Lib/HTTP/URLSessionWithUser.swift)